tekton/pipeline_run.go

```diff
@@ -102,6 +104,19 @@ func (r *ReleasePipelineRun) WithReleaseLabels(releaseName, releaseNamespace str
 	return r
 }
 
+// WithApplicationSnapshot adds a param containing the ApplicationSnapshot as a json string
+// to the release PipelineRun.
+func (r *ReleasePipelineRun) WithApplicationSnapshot(snapshot *v1alpha1.ApplicationSnapshot) *ReleasePipelineRun {
+	snapshotString, _ := json.Marshal(snapshot.Spec)
```

Comment:
Can we maybe add a comment here saying that we ignore the error as this should not happen when marshalling specs and that if we end up needing this we will need to pass the errors through the chain and something like a `Complete` function returning the final object and an error?